tests/setup-test.js

```diff
@@ -1,19 +1,71 @@
 'use strict';
 
 var tape = require('tape');
-var server = require('../example/server.js');
+// var Hoek = require('hoek');
+var init = require('../example/server.js').init;
+var tagsPool = require('../example/server.js').tagsPool;
+var tags = require('../example/tags.json');
+var server;
+
+// First check that the DB is empty
+tape('check that the DB is empty', function (t) {
+  tagsPool.connect(function (conErr, client, done) {
+    if (conErr) {
+      t.fail();
+    }
+    t.equal(typeof client, 'object',
+      'We can connect to the db');
+
+    client.query('select * from tags;', function (dbErr, results) {
+      if (dbErr) {
+        return t.fail();
+      }
+
+      t.equal(typeof results.rows, 'object',
+        'we successfully connected to the tags table');
+
+      t.equal(results.rows.length, 0,
+        'The tags table has 0 rows');
+
+      t.end();
+
+      return done();
+    });
+  });
+});
+
+
+// set up server
+tape('set up server', function (t) {
+  init(2000, function (err, newServer) {
+    if (err) {
+      return t.fail();
+    }
+    server = newServer;
+
+    return t.end();
+  });
+});
+
 
 tape('check that plugin is being attached to the request object', function (t) {
   server.inject({
```

Comment:
If we say that only the init function will configure the pg pool and the server then we don't have to recreate the configuration on the tests and you just have to reuse the init function. I think we need to determine which tests we want, do we want to test the database or the endpoints of the app? This will determine which feature you need to run your tests. At the moment I think we can simplify by only using init in our tests. We can have a script which clear/initialise the database but I think we should focus more on the endpoints. I'm not sure it's clear, let me know